Yoast/wpseo-news
# Tests/WP: use PSR-4

## Context

* Improve dev experience 

## Summary

This PR can be summarized in the following changelog entry:

* Tests: use PSR-4

## Relevant technical choices:

### Tests/WP: remove redundant @requires tag

As PHP 7.2 is the minimum supported version, this `@requires` and the associated comment are now redundant.


### Tests/WP: use PSR-4

* Move the existing WP-based unit tests to a `tests/WP` subdirectory and ensure subdirectory names use "ProperCase".
* Namespace all test files and remove file docblocks.
    Includes making all references to global functions and constants fully qualified and importing all used classes.
* Ensure all test class file names match the name of the test class.
* Remove prefixes from the test class names which are not needed in namespaced classes (`WPSEO_News`) and where relevant, update references to those classes in the test files.
* Change the Composer `autoload-dev` directive to use PSR-4 autoloading for these tests.
* Updates the names for the scripts to run these tests in the `composer.json` file.
    Note: the "old" `integration-test` and `integration-coverage` Composer scripts will still be available for the time being to facilitate the switch over and give the teams a little time to get used to the new script names.
* Rename the `phpunit-integration.xml.dist` file to `phpunit-wp.xml.dist` and update it to match the other changes.
* Update the terminology used in the GH actions `test` workflow.
* Update the `.gitattributes` and `.gitignore` files to match.
* Update the `phpcs.xml.dist` file to match.

Includes removing the `framework` subdirectory and moving the base `TestCase` class to the root directory for these tests.

## Test instructions

### Test instructions for the acceptance test before the PR gets merged
This PR can be acceptance tested by following these steps:

* _N/A_ If the tests pass, we're good.
